backend/s3/s3.go

```diff
@@ -985,6 +986,11 @@ func s3Connection(opt *Options) (*s3.S3, *session.Session, error) {
 			}),
 			ExpiryWindow: 3 * time.Minute,
 		},
+
+		// Pick up IAM role if we are in EKS
+		&stscreds.WebIdentityRoleProvider{
```

Comment:
Does creating this struct this way really work? It has quite a lot of non-public fields that are set when calling `NewWebIdentityRoleProvider`. As you can see that state is being used in the `Retrieve` function: https://github.com/aws/aws-sdk-go/blob/master/aws/credentials/stscreds/web_identity_provider.go#L67

Reply:
@tirsen - is this causing you a problem?

Want to send a PR to fix?

Reply:
Yeah EKS role authentication doesn't work. I've checked all the env vars of the pod but everything seems to be in good order. Tracing through the code paths of `WebIdentityRoleProvider` I can see that the env vars will not be read if that struct is created that way.

Here is a fix:
https://github.com/rclone/rclone/pull/4023